agent/php_nrini.c

```diff
@@ -222,6 +223,53 @@
 
 ZEND_DECLARE_MODULE_GLOBALS(newrelic)
 
+#define NR_INI_PREFIX_LEN 9
+
+char* nr_ini_to_env(const char* ini_name) {
+  char* env_name = NULL;
+  char* buf = NULL;
+  char* ini_upper = NULL;
+  const char* NR_PREFIX_STR = "NEW_RELIC";
+  int ini_len = nr_strlen(ini_name);
+
+  // 'newrelic.' is 9 characters - anything less should be rejected.
+  if (NULL == ini_name || nr_strlen(ini_name) <= NR_INI_PREFIX_LEN) {
+    return NULL;
+  }
+
+  // ini value should start with the 'newrelic.' prefix
+  if (0 != nr_stridx(ini_name, "newrelic.")) {
+    return NULL;
+  }
+
+  // algorithm:
+  // 1. uppercase string
+  // 2. copy the entire string after 'newrelic.'
+  // 3. iterate through new string and swap each '.' with '_'
+  // 4. snprintf append the string to 'NEW_RELIC_'
+  // 5. return result
+
+  ini_upper = nr_string_to_uppercase(ini_name);
+
+  buf = (char*)nr_malloc(ini_len - NR_INI_PREFIX_LEN + 1);
+  nr_strcpy(buf, ini_upper + NR_INI_PREFIX_LEN);
+
+  for (int i = 0; i < nr_strlen(buf); i++) {
+    if (buf[i] == '.') {
```

Comment:
If you change the allocation of `buf` to use `nr_zalloc()` and then change this check to 

```
if ('.' == buf[i] && (0 == i || '_' != buf[i-1])) {
```
then this will handle consecutive underscores properly.  By switching to `nr_zalloc()` then as you build the string it will always be zero terminated, accounting for this change might make the string shorter than the original.

Reply:
incorporated this feedback here: d58b00e68f8ab2587fa3cfee4d4cde5b64f44e88